scripts/license/add_license_header_test.go

```diff
@@ -0,0 +1,153 @@
+	// make a temporary folder
+	tmpDir := t.TempDir()
+
+	// make a temporary file in that folder
+	tmpFileName := filepath.Join(tmpDir, "test_license")
+	require.NoError(t, os.WriteFile(tmpFileName, []byte(addPrefix(licenseHeader, "//")+"\npackage main"), 0660))
+
+	originalContent, err := os.ReadFile(tmpFileName)
+	require.NoError(t, err)
+
+	// create a copy of the file with the same content
+	copyName := filepath.Join(tmpDir, "test_geth_license_copy.go")
+	require.NoError(t, os.WriteFile(copyName, originalContent, 0440))
+
+	require.NoError(t, processFiles(tmpDir, ".go", "//", licenseHeader, false, false))
+
+	contentAfter, err := os.ReadFile(copyName)
+	require.NoError(t, err)
+	require.Equal(t, originalContent, contentAfter)
+}
+
+func Test_Replaces_OldLicenseHeader(t *testing.T) {
+	// make a temporary folder
+	tmpDir := t.TempDir()
+	// make a temporary file in that folder
+	tmpFileName := filepath.Join(tmpDir, "test_license.go")
+	// write a sample license header to the file
+	oldLicense := `Copyright 2024 Sonic Operations Ltd
+				   This file is part of some old version
+				   of the Sonic Client`
+	originalContent := []byte(addPrefix(oldLicense, "//") + "\npackage main\n")
+
+	require.NoError(t, os.WriteFile(tmpFileName, originalContent, 0660))
+
+	require.NoError(t, processFiles(tmpDir, ".go", "//", licenseHeader, false, false))
+
+	content, err := os.ReadFile(tmpFileName)
+	require.NoError(t, err)
+	require.Contains(t, string(content), addPrefix(licenseHeader, "//"))
+
+	require.NotContains(t, string(content), oldLicense)
+}
+
+func Test_Adds_LicenseHeader(t *testing.T) {
+	// make a temporary folder
+	tmpDir := t.TempDir()
+	// make a temporary file in that folder
+	tmpFileName := filepath.Join(tmpDir, "test_license.go")
+	// write a file without license header
+	require.NoError(t, os.WriteFile(tmpFileName, []byte("package main\n\nfunc main() {}\n"), 0660))
+
+	require.NoError(t, processFiles(tmpDir, ".go", "//", licenseHeader, false, false))
+
+	content, err := os.ReadFile(tmpFileName)
+	require.NoError(t, err)
+	extendLicenseHeader := addPrefix(licenseHeader, "//")
+	require.Contains(t, string(content), extendLicenseHeader)
+}
+
+func Test_Detects_DoubleHeader(t *testing.T) {
+	// make a temporary folder
+	tmpDir := t.TempDir()
+	// make a temporary file in that folder
+	tmpFileName := filepath.Join(tmpDir, "test_license.go")
+	doubleHeaderString := addPrefix(licenseHeader, "//") +
+		addPrefix(licenseHeader, "//") +
+		"\npackage main"
+	require.NoError(t, os.WriteFile(tmpFileName, []byte(doubleHeaderString), 0660))
+
+	// Check for double license headers
+	require.Error(t, checkDoubleHeader(tmpFileName, "//"))
+}
+
+func Test_OnlyOneEmptyLineAfterHeader(t *testing.T) {
+	// make a temporary folder
+	tmpDir := t.TempDir()
+	// make a temporary file in that folder
+	tmpFileName := filepath.Join(tmpDir, "test_license.go")
+
+	fileWithHeader := addPrefix(licenseHeader, "//") + "\npackage main\nfunc main() {}\n"
```

Comment:
It looks like this test is preparing a file with a proper license header. So the `processFiles` should not have any affect.

A more interesting test case would be if you start with a file with a few empty lines on top, an empty file, or a file with a doc-string on top.